staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -128,6 +156,27 @@ func (c *Config) SkipComplete() completedConfig {
 	return completedConfig{c}
 }
 
+func (r *aggregatorRoutingGenerator) GetServiceResolver(defaultHost string) ServiceResolver {
+	return &aggregatorRouting{
+		routing:     r,
+		defaultHost: defaultHost,
+	}
+}
+
+func (r *aggregatorRouting) ResolveEndpoint(namespace, name string) (*url.URL, error) {
+	if r.routing.enabled {
```

Comment:
This global if here indicates that we actually have two dissimilar types, both of which implement the interface.  One that wants to resolve endpoints itself and one that wants to use a service IP.  Let's separate the two.